src/main/java/unicash/ui/StatusBarFooter.java

```diff
@@ -1,29 +1,106 @@
+ * balance indicator.
  */
 public class StatusBarFooter extends UiPart<Region> {
 
     private static final String FXML = "StatusBarFooter.fxml";
+    private static final String BALANCE_FORMAT = "Rolling Balance: $%.2f";
 
     @FXML
     private Label saveLocationStatus;
 
+    @FXML
+    private Label balanceIndicator;
+
+    private ObservableList<Transaction> transactions;
+
     /**
-     * Creates a {@code StatusBarFooter} with the given {@code Path}
-     * indicating the current Data Source.
+     * Creates a {@code StatusBarFooter} with the given {@code Path} indicating the
+     * current Data Source and a given {@code ObservableList} of transactions to
+     * facilitate the updating of balance and keeping a balance counter.
      */
-    public StatusBarFooter(Path saveLocation) {
+    public StatusBarFooter(Path saveLocation,
+                           ObservableList<Transaction> transactionList) {
         super(FXML);
         saveLocationStatus.setText("Data source -> " + Paths.get(".").resolve(saveLocation));
+        this.transactions = transactionList;
+
+        /*
+         * This initiates the first indication of balance upon application start,
+         * before the method is added as a listener.
+         */
+        updateBalance(transactions);
+
+        /*
+         * The {@code updateBalance} method is added as a listener, which means that every
+         * time the transactions list is updated, the balanceIndicator would be updated
+         * dynamically. This is done with a {@code ListChangeListener} that reports a list
+         * of changes done to an {@code ObservableList}
+         */
+        this.transactions.addListener((
+                ListChangeListener.Change<? extends Transaction> c) -> {
+            updateBalance(transactions);
+        });
+    }
+
+    /**
+     * Handles the updating of balanceIndicator in the status bar. If the balance
+     * falls below 0, or goes into a negative value, the font style is set to the
+     * universal red color style, and green otherwise.
+     *
+     * </p> It must be noted that given the way {@code MainWindow} interacts with
+     * {@code Logic}, all Commands that result in a change to the filtered list on
+     * display, such as any filter (find, get_total, etc.) commands will result in
+     * the balance being tabulated only for those specific transactions contained in
+     * the current TransactionListPanel. This is the nature of the observable list,
+     * and a persistent universal balance indicator would require modifications to
+     * the interaction between {@code MainWindow} and {@Logic} minimally.
+     *
+     * @param transactions the input transactions list to be monitored.
+     */
+    @FXML
+    public void updateBalance(ObservableList<Transaction> transactions) {
+        double balance = 0.0;
+
+        for (Transaction t : transactions) {
+            if (t.getTypeString().equalsIgnoreCase("expense")) {
```

Comment:
Could we use the pre-assigned `TransactionType.EXPENSE.type` String here instead? This way if we do rename the enum string, we don't leave this to have undefined behavior